### PR TITLE
feat!: functional programming and typescript migration

### PR DESCRIPTION
**Pull request by jobo322**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,177 +0,0 @@
-# Changelog
```
Comment (targos) [inline CHANGELOG.md:1]: please keep the changelog

File: README.md [hunk 2]
```diff
@@ -2,8 +2,10 @@
 
 [![NPM version][npm-image]][npm-url]
 [![build status][ci-image]][ci-url]
+[![Test coverage][codecov-image]][codecov-url]
 [![npm download][download-image]][download-url]
 
+.
```
Comment (targos) [inline README.md:8]:
```suggestion
```

File: README.md [hunk 3]
```diff
@@ -51,20 +53,47 @@ let shapeGenerator = getShapeGenerator('lorentzian', {factor: 3.5, sd: 500});
...
-[download-image]: https://img.shields.io/npm/dm/ml-peak-shape-generator.svg
-[download-url]: https://www.npmjs.com/package/ml-peak-shape-generator
+[npm-image]: https://img.shields.io/npm/v/ml-ml-peak-shape-generator.svg
+[npm-url]: https://www.npmjs.com/package/ml-ml-peak-shape-generator
+[ci-image]: https://github.com/mljs/ml-peak-shape-generator/workflows/Node.js%20CI/badge.svg?branch=main
+[ci-url]: https://github.com/mljs/ml-peak-shape-generator/actions?query=workflow%3A%22Node.js+CI%22
+[codecov-image]: https://img.shields.io/codecov/c/github/mljs/ml-peak-shape-generator.svg
+[codecov-url]: https://codecov.io/gh/mljs/ml-peak-shape-generator
+[download-image]: https://img.shields.io/npm/dm/ml-ml-peak-shape-generator.svg
+[download-url]: https://www.npmjs.com/package/ml-ml-peak-shape-generator
```
Comment (targos) [inline README.md:99]:
```suggestion
[npm-image]: https://img.shields.io/npm/v/ml-peak-shape-generator.svg
[npm-url]: https://www.npmjs.com/package/ml-peak-shape-generator
[ci-image]: https://github.com/mljs/peak-shape-generator/workflows/Node.js%20CI/badge.svg?branch=main
[ci-url]: https://github.com/mljs/peak-shape-generator/actions?query=workflow%3A%22Node.js+CI%22
[codecov-image]: https://img.shields.io/codecov/c/github/mljs/peak-shape-generator.svg
[codecov-url]: https://codecov.io/gh/mljs/peak-shape-generator
[download-image]: https://img.shields.io/npm/dm/ml-peak-shape-generator.svg
[download-url]: https://www.npmjs.com/package/ml-peak-shape-generator
```

File: README.md [hunk 4]
```diff
@@ -51,20 +53,47 @@ let shapeGenerator = getShapeGenerator('lorentzian', {factor: 3.5, sd: 500});
...
 
 ```js
-import { Gaussian } from 'ml-peak-shape-generator';
-const func = Gaussian.fct(x - mean, fwhm);
+import { gaussian } from 'ml-peak-shape-generator';
+const func = gaussian.fct(fwhm, x - mean);
 
 ```
 
+In that way you can currified the function with some thing like:
```
Comment (targos) [inline README.md:61]: I'm not sure this section is really useful.

File: .github/workflows/typedoc.yml [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+        uses: zakodium/typedoc-action@v2
+        with:
+          entry: ${{ env.ENTRY_FILE }}
+      - name: Deploy to GitHub pages
+        uses: JamesIves/github-pages-deploy-action@releases/v4
+        with:
+          token: ${{ secrets.BOT_TOKEN }}
+          branch: gh-pages
+          folder: docs
+          clean: true
```
Comment (targos) [inline .github/workflows/typedoc.yml:63]: the file contents are duplicated

**Comment by targos** [comment]
Did you copy the structure of an existing project to create this?
I'm asking because this is a missing tsconfig for ESM browser output.

**Comment by jobo322** [comment]
> Did you copy the structure of an existing project to create this?
> I'm asking because this is a missing tsconfig for ESM browser output.

I did use the cheminfo-tools to generate a new typescript project, when all are running well I did move the files to the main project.

File: README.md (hunk 2 above)
Comment (jobo322) [reply README.md]: it was removed

**Comment by targos** [comment]
> > Did you copy the structure of an existing project to create this?
> > I'm asking because this is a missing tsconfig for ESM browser output.
> 
> I did use the cheminfo-tools to generate a new typescript project, when all are running well I did move the files to the main project.

I'm not sure what you mean?
The only tool I know we use to generate projects is `yo`, and our generator includes a `tsconfig.esm.json` file (and related config in the package.json): https://github.com/cheminfo/generator-cheminfo/blob/main/generators/typescript/templates/tsconfig.esm.json

**Comment by jobo322** [comment]
> > > Did you copy the structure of an existing project to create this?
> > > I'm asking because this is a missing tsconfig for ESM browser output.
> > 
> > 
> > I did use the cheminfo-tools to generate a new typescript project, when all are running well I did move the files to the main project.
> 
> I'm not sure what you mean?
> The only tool I know we use to generate projects is `yo`, and our generator includes a `tsconfig.esm.json` file (and related config in the package.json): https://github.com/cheminfo/generator-cheminfo/blob/main/generators/typescript/templates/tsconfig.esm.json

yes, I did it with yo cheminfo:typescript

**Pull request closed without merging**